docs/scos/dev/guidelines/keeping-a-project-upgradable/upgradability-guidelines/private-api-is-used-on-the-project-level.md

```diff
@@ -128,3 +134,177 @@ class CustomerReader extends SprykerCustomerReader implements CustomerReaderInte
 ```
 
 After the fix re-evaluate the code. The same error shouldn’t be returned.
+
+
+## PrivateApi:Persistence
+
+Repository and entity manager on project level can use the factory but only methods that were declared on project level
```

Comment:
Here we need to formulate it in another way:

Maybe something like:
```
It is allowed to use factory classes inside of repositories and entity manager classes on project level, but only the methods from these classes, that were declared on project level.
```

Reply:
++

Replaced with the suggested structure